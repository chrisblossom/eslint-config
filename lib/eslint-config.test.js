'use strict';

const path = require('path');
const readDirDeep = require('read-dir-deep');
const Eslint = require('eslint').CLIEngine;

const files = readDirDeep
    .sync(path.resolve(__dirname, '__sandbox__/base/'))
    .map((file) => path.resolve(__dirname, '__sandbox__/base/', file));

const eslint = new Eslint({
    baseConfig: require('./eslint-config'),
    useEslintrc: false,
    ignore: false,
});

test('lints vanilla javascript files', () => {
    const result = eslint.executeOnFiles(files).results;

    expect(result).toMatchSnapshot();
});
