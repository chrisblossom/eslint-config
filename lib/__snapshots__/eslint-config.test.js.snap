// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lints vanilla javascript files 1`] = `
Array [
  Object {
    "errorCount": 1,
    "filePath": "<PROJECT_ROOT>/lib/__sandbox__/base/a.js",
    "fixableErrorCount": 0,
    "fixableWarningCount": 0,
    "messages": Array [
      Object {
        "column": 7,
        "endColumn": 13,
        "endLine": 1,
        "line": 1,
        "message": "'unused' is assigned a value but never used.",
        "nodeType": "Identifier",
        "ruleId": "no-unused-vars",
        "severity": 2,
      },
    ],
    "source": "const unused = 1;

function a() {
}

export { a };
",
    "warningCount": 0,
  },
  Object {
    "errorCount": 1,
    "filePath": "<PROJECT_ROOT>/lib/__sandbox__/base/b.js",
    "fixableErrorCount": 0,
    "fixableWarningCount": 0,
    "messages": Array [
      Object {
        "column": 1,
        "endColumn": 18,
        "endLine": 7,
        "line": 7,
        "message": "Prefer named exports.",
        "nodeType": "ExportDefaultDeclaration",
        "ruleId": "import/no-default-export",
        "severity": 2,
      },
    ],
    "source": "import { a } from './a';

function b() {
    a();
}

export default b;
",
    "warningCount": 0,
  },
]
`;
